cassandra-executor/src/test/java/io/mesosphere/mesos/frameworks/cassandra/CassandraExecutorTest.java

```diff
@@ -0,0 +1,267 @@
+package io.mesosphere.mesos.frameworks.cassandra;
+
+import org.apache.cassandra.service.ActiveRepairService;
+import org.apache.mesos.Protos;
+import org.junit.Test;
+
+import java.util.Collections;
+import java.util.List;
+
+import static org.junit.Assert.*;
+
+public class CassandraExecutorTest {
+    CassandraExecutor executor;
+    MockExecutorDriver driver;
+    TestObjectFactory objectFactory;
+    Protos.TaskID taskIdMetadata;
+    Protos.TaskID taskIdServer;
+
+    @Test
+    public void testStartup() throws Exception {
+        cleanState();
+
+        startServer();
+    }
+
+    @Test
+    public void testExecutorRepair() throws Exception {
+        cleanState();
+
+        startServer();
+
+        repairJob();
+
+        repairJob();
+    }
+
+    @Test
+    public void testExecutorCleanup() throws Exception {
+        cleanState();
+
+        startServer();
+
+        cleanupJob();
+
+        cleanupJob();
+    }
+
+    @Test
+    public void testExecutorRepairCleanup() throws Exception {
+        cleanState();
+
+        startServer();
+
+        repairJob();
+
+        cleanupJob();
+
+        repairJob();
+
+        cleanupJob();
+    }
+
+    private void cleanupJob() {
+        CassandraFrameworkProtos.ClusterJobType jobType = CassandraFrameworkProtos.ClusterJobType.CLEANUP;
+
+        assertNull(executor.getCurrentJob());
```

Comment:
Please replace default junit assertions with assertj for extended functionality and easier readability.

http://joel-costigliola.github.io/assertj/